{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9d40d3c7",
   "metadata": {},
   "source": [
    "### NGS Data Analysis: Part_1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ea7130e",
   "metadata": {},
   "source": [
    "**IP ADDRESS:** 14.139.82.215\n",
    "\n",
    "Windows users can connect through Putty \n",
    "\n",
    "**Ubuntu/Mac Users can utilize:**ssh -X user_name@14.139.82.215\n",
    "\n",
    "Difficulty Level - Basic to Intermediate\n",
    "\n",
    "\n",
    "### LIST OF COMMANDS TO BE PRACTISED.\n",
    "\n",
    "##### Creating a new folder \n",
    " mkdir ngc_workshop\n",
    "\n",
    "##### Creating Multiple folders in one go -\n",
    " mkdir ngc cdfd\n",
    "\n",
    "##### Creating an empty file\n",
    " touch dna\n",
    "\n",
    "##### Creating multiple empty files \n",
    " touch mrna trna\n",
    "\n",
    "##### Check the contents of the directory\n",
    " ls\n",
    "\n",
    "##### Changing the directory \n",
    " cd Day1\n",
    "\n",
    "##### Check the contents of Day1 directory\n",
    " ls\n",
    "\n",
    "##### Display hidden files and folders/directories\n",
    " ls -a\n",
    "\n",
    "##### Create a new directory by the name of practise\n",
    " mkdir practise\n",
    "\n",
    "##### Enter the new directory “practise”\n",
    " cd practise\n",
    "\n",
    "#####  Create a file named “hello_world” and enter some text in it \n",
    " cat > hello_world\n",
    "\n",
    "My name is Avinash.\n",
    "I am learning Linux\n",
    "\n",
    "Press Enter followed by Ctrl+D to come out of the text editor mode.\n",
    "\n",
    "##### Create two files by the name of “numbers” and “alphabets”\n",
    " cat > numbers \n",
    "\n",
    "1\n",
    "2\n",
    "3\n",
    "4\n",
    "5\n",
    "6\n",
    "7\n",
    "8\n",
    "9\n",
    "10\n",
    "\n",
    "Enter\n",
    "Ctrl+D\n",
    "\n",
    " cat > alphabets \n",
    "\n",
    "a\n",
    "b\n",
    "c\n",
    "d\n",
    "e\n",
    "f\n",
    "g\n",
    "h\n",
    "i\n",
    "j\n",
    "\n",
    "Enter\n",
    "Ctrl+D\n",
    "\n",
    "##### Display the contents of a text file \n",
    " cat hello_world\n",
    "\n",
    "##### Display the contents of multiple text files together \n",
    " cat numbers alphabets.\n",
    "\n",
    "##### Count the number of lines in “alphabets” \n",
    " wc -l alphabets \n",
    "\n",
    "##### Count the number of words in “hello_world”\n",
    " wc -w hello_world\n",
    "\n",
    "##### Zip the file “numbers”\n",
    " gzip numbers\n",
    "\n",
    "##### Use the “Piping” feature to display file content of “hello_world” & “alphabets” together and zip the combined content on the go.\n",
    " cat hello_world alphabets | gzip > compressed.gz\n",
    "\n",
    "##### Unzip the “compressed.gz” file\n",
    " gunzip compressed.gz\n",
    "\n",
    "##### Move one folder back \n",
    " cd ..\n",
    "\n",
    "##### gzip works on files only, for zipping the folders -\n",
    "\n",
    " tar -zvcf compressed.tar.gz practise/\n",
    "\n",
    "##### Create an empty folder “genomics”\n",
    " mkdir genomics\n",
    "\n",
    "##### Delete the “genomics” \n",
    " rmdir genomics\n",
    "\n",
    "##### Delete the non-empty folder “practise”\n",
    " rm -rf practise\n",
    "\n",
    "##### Check the contents of the folder\n",
    " ls\n",
    "\n",
    "##### Enter the “Linux” folder\n",
    " cd Linux\n",
    "\n",
    "##### Check the contents of “Linux” folder\n",
    " ls \n",
    "\n",
    "**FASTA FILES:** FASTA format is a text-based format for representing either nucleotide sequences or amino acid (protein) sequences, in which nucleotides or amino acids are represented using single-letter codes.Each sequence will have a “>” symbol in the beginning and it is restricted to that location only.\n",
    "\n",
    "**FASTQ FILES:** A FASTQ file is a text file that contains the sequence data from the clusters that pass filter on a flow cell (for more information on clusters passing filter. Each FASTQ file consists of reads, each read in turn consisting of 4 lines.Every read will have “@” symbol in the beginning.However that is not the only position of the “@” symbol in the entire file.\n",
    "\n",
    "##### Display first 10 lines of the demo.fasta file\n",
    " head -10 demo.fasta file\n",
    "\n",
    "##### Display the last 10 lines of demo.fasta file\n",
    " tail -10 demo.fasta file.\n",
    "\n",
    "Display the number of lines in the “reads_1.fq” file\n",
    "Display the number of lines in the “reads_2.fq” file\n",
    "Find the number of sequences in “reads_1.fq” & “reads_2.fq”\n",
    "Display the first 12 lines of reads_1.fq\n",
    "Display the last 12 lines of reads_2.fq\n",
    "\n",
    "##### Display the number of lines that have @ symbol in each line \n",
    " grep “@” reads_1.fq\n",
    "\n",
    "#### Count the number of lines that have @ symbol in each line\n",
    " grep “@” reads_1.fq -c\n",
    "\n",
    "##### Calculate the number of sequences in the “demo.fasta” file.\n",
    "Hint - Head count (no. of heads == no of people). Look for unique identifier\n",
    "head -12 reads_1.fq\n",
    "wc -l reads_1.fq\n",
    "\n",
    "\n",
    "##### Replace the occurrence of “@” in the reads_1.fastq\n",
    " sed ‘s/@/>/g’ reads_1.fastq\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b50580f2",
   "metadata": {},
   "source": [
    "**Thank you** "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
